cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/MDSStreamMetaStore.java

```diff
@@ -118,14 +122,22 @@ public Boolean apply(StreamMds mds) throws Exception {
   }
 
   @Override
-  public List<StreamSpecification> listStreams() throws Exception {
-    return txnl.executeUnchecked(new TransactionExecutor.Function<StreamMds, List<StreamSpecification>>() {
-      @Override
-      public List<StreamSpecification> apply(StreamMds mds) throws Exception {
-        return mds.streams.list(new MetadataStoreDataset.Key.Builder().add(TYPE_STREAM).build(),
-                                StreamSpecification.class);
-      }
-    });
+  public Multimap<NamespaceMeta, StreamSpecification> listStreams() throws Exception {
+    return txnl.executeUnchecked(
+      new TransactionExecutor.Function<StreamMds, Multimap<NamespaceMeta, StreamSpecification>>() {
+        @Override
+        public Multimap<NamespaceMeta, StreamSpecification> apply(StreamMds mds) throws Exception {
+          ImmutableMultimap.Builder<NamespaceMeta, StreamSpecification> builder = ImmutableMultimap.builder();
+          List<NamespaceMeta> namespaces =
```

Comment:
We should be able to do this in a better way later, without having to query all namespaces, I think. If we are storing the ``accountId`` in the ``rowKey``, then we should be able to identify the namespace from it later. Separate PR for 2.8.0 though, maybe while namespacing streams. Can you report a jira and add a TODO for later? Feel free to assign to me.